pkg/testing/cli-suite-builder.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 VMware
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testing
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/cartographer/pkg/templates"
+)
+
+type testInfo struct {
+	Name                 *string                   `yaml:"name"`
+	Description          *string                   `yaml:"description"`
+	Template             *string                   `yaml:"template"`
+	Workload             *string                   `yaml:"workload"`
+	Expected             *string                   `yaml:"expected"`
+	Ytt                  *string                   `yaml:"ytt"`
+	SupplyChainInputs    *templates.Inputs         `yaml:"supplyChainInputs"`
+	BlueprintParams      []v1alpha1.BlueprintParam `yaml:"blueprintParams"`
+	Focus                *bool                     `yaml:"focus"`
+	IgnoreMetadata       *bool                     `yaml:"ignoreMetadata"`
+	IgnoreOwnerRefs      *bool                     `yaml:"ignoreOwnerRefs"`
+	IgnoreLabels         *bool                     `yaml:"ignoreLabels"`
+	IgnoreMetadataFields []string                  `yaml:"ignoreMetadataFields"`
+}
+
+const (
+	templateDefaultFilename  = "template.yaml"
+	workloadDefaultFilename  = "workload.yaml"
+	expectedDefaultFilename  = "expected.yaml"
+	yttValuesDefaultFilename = "ytt-values.yaml"
+)
+
+func buildTestSuite(testCase TemplateTestCase, directory string) (TemplateTestSuite, error) {
+	info, err := populateInfo(directory)
+	if err != nil {
+		return nil, fmt.Errorf("populate info: %w", err)
+	}
+
+	testCase.Given.TemplateFile = replaceIfFound(testCase.Given.TemplateFile, directory, templateDefaultFilename, info.Template)
+	testCase.Given.WorkloadFile = replaceIfFound(testCase.Given.WorkloadFile, directory, workloadDefaultFilename, info.Workload)
+	testCase.Expect.ExpectedFile = replaceIfFound(testCase.Expect.ExpectedFile, directory, expectedDefaultFilename, info.Expected)
+
+	yttFile := ""
+	if testCase.Given.YttFiles != nil {
```

Comment:
maybe adding a _"hey, having `ytt` available under $PATH is a pre-requisite for this"_ in the [YTT preprocessing](https://github.com/vmware-tanzu/cartographer-site/blob/6515642bac58d759ae48328f958490b3c276b8c0/content/docs/development/testing-templates.md#ytt-preprocessing-file) section of the docs too

Reply:
When ytt is not installed, users will now see the message `test requires ytt, but ytt was not found in path` and skips trace information.

Example, the output for the command above (when ytt not installed) is now:
```console
$ go run ./cmd/cartotest --directory ./tests/templates -v

DEBU[0000] populate info failed, did not find tests/templates/info.yaml 

PASS: tests/templates/deliverable/regular-template
PASS: tests/templates/kpack image-template
FAIL: tests/templates/deliverable/ytt-preprocess
Description: template is incomplete without being formatted with ytt and the ytt-values.yaml file
Error: test requires ytt, but ytt was not found in path
FAIL: tests/templates/deliverable/ytt-template
Name: YTT Template Test
Description: The template's spec.ytt field exists instead of a spec.template field
Error: test requires ytt, but ytt was not found in path
FAIL
exit status 1
```

I've also added a section in the docs. Awaiting PR review here: https://github.com/vmware-tanzu/cartographer-site/pull/164